src/sentry/api/endpoints/organization_member_invite/reinvite.py

```diff
@@ -0,0 +1,156 @@
+from typing import Any
+
+from django.db import router, transaction
+from rest_framework import status
+from rest_framework.exceptions import PermissionDenied
+from rest_framework.request import Request
+from rest_framework.response import Response
+
+from sentry import audit_log, features, ratelimits
+from sentry.api.api_owners import ApiOwner
+from sentry.api.api_publish_status import ApiPublishStatus
+from sentry.api.base import region_silo_endpoint
+from sentry.api.bases.organization import OrganizationEndpoint, OrganizationPermission
+from sentry.api.endpoints.organization_member_invite import MISSING_FEATURE_MESSAGE
+from sentry.api.exceptions import ResourceDoesNotExist
+from sentry.api.serializers import serialize
+from sentry.api.serializers.rest_framework.organizationmemberinvite import (
+    OrganizationMemberReinviteRequestValidator,
+)
+from sentry.models.organization import Organization
+from sentry.models.organizationmemberinvite import OrganizationMemberInvite
+from sentry.utils import metrics
+
+ERR_EXPIRED = "You cannot resend an expired invitation without regenerating the token."
+ERR_INSUFFICIENT_SCOPE = "You are missing the member:admin scope."
+ERR_INVITE_UNAPPROVED = "You cannot resend an invitation that has not been approved."
+ERR_MEMBER_INVITE = "You cannot resend an invitation that was sent by someone else."
+ERR_RATE_LIMITED = "You are being rate limited for too many invitations."
+
+
+class MemberReinvitePermission(OrganizationPermission):
+    scope_map = {"PUT": ["member:invite", "member:write", "member:admin"]}
+
+
+@region_silo_endpoint
+class OrganizationMemberReinviteEndpoint(OrganizationEndpoint):
+    publish_status = {
+        "PUT": ApiPublishStatus.EXPERIMENTAL,
+    }
+    owner = ApiOwner.ENTERPRISE
+    permission_classes = (MemberReinvitePermission,)
+
+    def convert_args(
+        self,
+        request: Request,
+        member_invite_id: str,
+        organization_id_or_slug: str | int | None = None,
+        *args: Any,
+        **kwargs: Any,
+    ) -> tuple[tuple[Any, ...], dict[str, Any]]:
+        args, kwargs = super().convert_args(request, organization_id_or_slug, *args, **kwargs)
+
+        try:
+            kwargs["invited_member"] = OrganizationMemberInvite.objects.get(
+                id=int(member_invite_id)
+            )
+        except OrganizationMemberInvite.DoesNotExist:
+            raise ResourceDoesNotExist
+        return args, kwargs
+
+    def _reinvite(
+        self,
+        request: Request,
+        organization: Organization,
+        invited_member: OrganizationMemberInvite,
+        regenerate: bool,
+    ) -> Response:
+        # for typing
+        if not request.user.is_authenticated:
+            return Response(status=status.HTTP_400_BAD_REQUEST)
```

Comment:
Is this check relevant to `_reinvite` method? Or is it more relevant in the POST method?